server/tests/src/test/java/org/infinispan/server/persistence/JdbcStringBasedCacheStoreIT.java

```diff
@@ -39,7 +40,9 @@ public class JdbcStringBasedCacheStoreIT {
     public void testFailover(Database database) throws Exception {
         JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC, database, false, true)
                 .setLockingConfigurations();
-        RemoteCache<String, String> cache = SERVERS.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+        RemoteCache<String, String> cache = SERVERS.hotrod()
```

Comment:
I'm really not a fan of having to modify all of these tests just for the DB use case. I think the way to go is to make the timeout configurable via a system property, then we can just apply this property when we execute the DB2 tests.

We can extend `HotRodTestClientDriver#applyDefaultConfiguration` so that system properties are applied to the builder via `withProperties()`. E.g. lookup create a `Properties` objection with all `infinispan.client.hotrod.*` on the path.

That way we can just add `-Dinfinispan.client.hotrod.socket_timeout=10000` when executing the tests with DB2.

Reply:
I decided to take slightly alter the solution and build the `infinispan.client.hotrod.*=xyz` properties from datatabase properties. That way it's not necessary to have an if statement in the the GitHub action definition and when somebody tries to run the test locally they won't have to add the property to the command line or be buffled why it's failing.